File: su-lim/프렌즈4블록.py
def grading(m, n, board):
    copy_board = [[0] * n for i in range(m)]
    
    # TODO 네모 위치에 1씩 더함, 나중에 0이 아닌 값의 수를 리턴해야함
    for i in range(n-1):
        for j in range(m-1):
            if( (board[i][j] == board[i][j+1]) and (board[i][j] == board[i+1][j])):
                if(board[i+1][j] == board[i+1][j+1]):
                    copy_board[i][j] += 1
                    copy_board[i][j+1] += 1
                    copy_board[i+1][j] += 1
                    copy_board[i+1][j+1] += 1

def solution(m, n, board):
    answer = 0
    return answer

File: hojin/week1/프렌즈4블록.cpp
#include <string>
#include <vector>
#include <iostream>
#include <cstring>
#include <queue>
#include <utility>

using namespace std;
using Pair = pair<int,int>;

int solution(int m, int n, vector<string> board) {
    int answer = 0, i = 0, j = 0, k = 0, check_idx = -1;
    queue<Pair> q;
    bool check = false;
    
    do{
        check = false;
        for(i=0;i<m-1;i++){
            for(j=0;j<n-1;j++){
                if(board[i][j] != '0' && board[i][j] == board[i][j+1] && board[i][j] == board[i+1][j] && board[i][j] == board[i+1][j+1]){
                    q.push({i,j});
                    q.push({i,j+1});
                    q.push({i+1,j});
                    q.push({i+1,j+1});
                    check = true;
                }
            }
        }

        while(!q.empty()){
            i = q.front().first;
            j = q.front().second;
            q.pop();
            if(board[i][j] != '0'){
                answer += 1;
            }
            board[i][j] = '0';
        }

        for(j=0;j<n;j++){
            check_idx = -1;
            for(i=m-1;i>=0;i--){
                if(board[i][j] == '0'){
                    for(k=i-1;k>=0;k--){
                        if(board[k][j] != '0'){
                            board[i][j] = board[k][j];
                            board[k][j] = '0';
                            break;
                        }
                    }
                }
            }
        }

    }while(check);

    return answer;
}

int main(){

    vector<string> board;
    //board.push_back("CCBDE");
    //board.push_back("AAADE");
    //board.push_back("AAABF");
    //board.push_back("CCBBF");
    board.push_back("AAAAAA");
    board.push_back("AABBAA");
    board.push_back("AABBAA");
    board.push_back("AAAAAA");
    board.push_back("AAACCA");
    board.push_back("AAACCA");


    cout << solution(6, 6, board) <<"\n";

    return 0;
}